Hardcodet.NotifyIcon.Wpf/Source/NotifyIconWpf/Interop/WindowMessageSink.cs

```diff
@@ -225,74 +225,74 @@ private void ProcessWindowMessage(uint msg, IntPtr wParam, IntPtr lParam)
         {
             if (msg != CallbackMessageId) return;
 
-            switch (lParam.ToInt32())
+            switch ((WindowsMessages)lParam.ToInt32())
             {
-                case 0x200:
+                case WindowsMessages.WM_MOUSEMOVE:
                     MouseEventReceived(MouseEvent.MouseMove);
                     break;
 
-                case 0x201:
+                case WindowsMessages.WM_LBUTTONDOWN:
                     MouseEventReceived(MouseEvent.IconLeftMouseDown);
                     break;
 
-                case 0x202:
+                case WindowsMessages.WM_LBUTTONUP:
                     if (!isDoubleClick)
                     {
                         MouseEventReceived(MouseEvent.IconLeftMouseUp);
                     }
                     isDoubleClick = false;
                     break;
 
-                case 0x203:
+                case WindowsMessages.WM_LBUTTONDBLCLK:
                     isDoubleClick = true;
                     MouseEventReceived(MouseEvent.IconDoubleClick);
                     break;
 
-                case 0x204:
+                case WindowsMessages.WM_RBUTTONDOWN:
                     MouseEventReceived(MouseEvent.IconRightMouseDown);
                     break;
 
-                case 0x205:
+                case WindowsMessages.WM_RBUTTONUP:
                     MouseEventReceived(MouseEvent.IconRightMouseUp);
                     break;
 
-                case 0x206:
+                case WindowsMessages.WM_RBUTTONDBLCLK:
                     //double click with right mouse button - do not trigger event
                     break;
 
-                case 0x207:
+                case WindowsMessages.WM_MBUTTONDOWN:
                     MouseEventReceived(MouseEvent.IconMiddleMouseDown);
                     break;
 
-                case 520:
+                case WindowsMessages.WM_MBUTTONUP:
                     MouseEventReceived(MouseEvent.IconMiddleMouseUp);
                     break;
 
-                case 0x209:
+                case WindowsMessages.WM_MBUTTONDBLCLK:
                     //double click with middle mouse button - do not trigger event
                     break;
 
-                case 0x402:
+                case WindowsMessages.WM_USER2:
                     var listener = BalloonToolTipChanged;
                     listener?.Invoke(true);
                     break;
 
-                case 0x403:
-                case 0x404:
+                case WindowsMessages.WM_USER3:
```

Comment:
I wouldn't know what they are officially called (a lot of that stuff I just scraped together from now obsolete docs, and trial and error). However, once that PR is through, those namings will be somewhat obscure. Suggestion: How about we call them `WM_0x404` or similar? That would be somewhat more self-documenting for someone who isn't as proficient at Win API as yourself.

Reply:
Those message ID's should be documented somewhere, and are probably for the control class the system tray is represented by. I will do some more digging, as I hate magic strings & numbers 🙄 

I was once trying to collect all the messages here: https://github.com/dapplo/Dapplo.Windows/blob/master/src/Dapplo.Windows.Messages/WindowsMessages.cs

That is part of a Win32 code project which contains code I separated from Greenshot. This should enable me to write my next "system" tool alone: A Clipboard manager, current working title "Dopy", but this will probably be obsolete by the time I finish it, considering what Microsoft is bringing -  https://github.com/dapplo/Dapplo.Dopy

Dapplo.Windows is not a perfect project, but it has some very cool abstractions for Win32 APIs. Like an observable for key strokes, allowing you to also specify complex hotkey combinations, you can also monitor registry or clipboard changes and way more.

This is also why I said that I might be able to optimize some Win32 calls, the following GetClassName implementation is a lot faster than using Marshalling of the string:
https://github.com/dapplo/Dapplo.Windows/blob/master/src/Dapplo.Windows.User32/User32Api.cs#L213

It still depends, if you are only calling it a few times, who cares.

Reply:
Oh, found them:
https://referencesource.microsoft.com/#System.Windows.Forms/winforms/Managed/System/WinForms/NativeMethods.cs,e75041b5218ff60b

```
        NIN_BALLOONSHOW = (WM_USER + 2),
        NIN_BALLOONHIDE = (WM_USER + 3),
        NIN_BALLOONTIMEOUT = (WM_USER + 4),
        NIN_BALLOONUSERCLICK = (WM_USER + 5),
```

Reply:
Which brought me here: https://github.com/dotnet/winforms/blob/master/src/System.Windows.Forms/src/System/Windows/Forms/NotifyIcon.cs#L823

Reply:
Will modify the enum accordingly, but not today.

Reply:
> Dapplo.Windows is not a perfect project, but it has some very cool abstractions for Win32 APIs. Like an observable for key strokes, allowing you to also specify complex hotkey combinations, you can also monitor registry or clipboard changes and way more.

That is quite interesting! Not necessarily for the NotifyIcon, but I assume a lot of projects that would work with an extended NotifyIcon might want to complement this with listeners you described (e.g. user performs an action such as a short cut, the software reacts and displays an update in the tray). That would make for a nice additional tutorial in the demo app for the next "big" release - let's keep that in mind :)

Reply:
> This is also why I said that I might be able to optimize some Win32 calls, the following GetClassName implementation is a lot faster than using Marshalling of the string:
> https://github.com/dapplo/Dapplo.Windows/blob/master/src/Dapplo.Windows.User32/User32Api.cs#L213
> 
> It still depends, if you are only calling it a few times, who cares.

If you stumble over them and optimization is an easy fix, and you're confident it doesn't introduce risk (e..g compatibility issues for some poor sobs that still have to use Win 8), by all means 🥇